packages/blade/src/components/Input/BaseInput/BaseInput.tsx

```diff
@@ -0,0 +1,472 @@
+     * **Note**: Few values are platform dependent and might not render on all the platforms
+     *
+     * `enter` is only available on web
+     * `previous` is only available on native android
+     */
+    keyboardReturnKeyType?: 'enter' | 'go' | 'done' | 'next' | 'previous' | 'search' | 'send';
+    /**
+     * **Web only**
+     *
+     * Hints browser to display an appropriate virtual keyboard
+     *
+     *
+     */
+    inputMode?: 'text' | 'search' | 'tel' | 'email' | 'url';
+    /**
+     * determines what autoComplete suggestion type to show
+     *
+     * Internally it'll render platform specific attributes:
+     *
+     * - web: `autocomplete`
+     * - iOS: `textContentType`
+     * - android: `autoComplete`
+     *
+     */
+    autoCompleteSuggestionType?:
+      | 'none'
+      | 'name'
+      | 'email'
+      | 'username'
+      | 'password'
+      | 'newPassword'
+      | 'oneTimeCode'
+      | 'telephone'
+      | 'postalCode'
+      | 'countryName'
+      | 'creditCardNumber'
+      | 'creditCardCSC'
+      | 'creditCardExpiry'
+      | 'creditCardExpiryMonth'
+      | 'creditCardExpiryYear';
+  };
+
+const autoCompleteSuggestionTypeValues = [
+  'none',
+  'name',
+  'email',
+  'username',
+  'password',
+  'newPassword',
+  'oneTimeCode',
+  'telephone',
+  'postalCode',
+  'countryName',
+  'creditCardNumber',
+  'creditCardCSC',
+  'creditCardExpiry',
+  'creditCardExpiryMonth',
+  'creditCardExpiryYear',
+];
+
+const useInput = ({
+  value,
+  defaultValue,
+  onChange,
+  onBlur,
+}: Pick<BaseInputProps, 'value' | 'defaultValue' | 'onChange' | 'onBlur'>): {
+  handleOnChange: HandleOnEvent;
+  handleOnBlur: HandleOnEvent;
+} => {
+  if (value && defaultValue) {
+    throw new Error(
+      `[Blade: Input]: Either 'value' or 'defaultValue' shall be passed. This decides if the input field is controlled or uncontrolled`,
+    );
+  }
+
+  const handleOnChange: HandleOnEvent = React.useCallback(
+    ({ name, value }) => {
+      let _value = '';
+
+      if (getPlatformType() === 'react-native' && typeof value === 'string') {
+        _value = value;
+      } else if (typeof value !== 'string') {
+        // it's weird but TS forced me to write this much code where I could have just done "getPlatformType() === 'react-native' ? value : value?.target.value" :(
+        _value = value?.target.value ?? '';
+      }
+
+      onChange?.({
+        name,
+        value: _value,
+      });
+    },
+    [onChange],
+  );
+
+  const handleOnBlur: HandleOnEvent = React.useCallback(
+    ({ name, value }) => {
+      let _value = '';
+
+      if (getPlatformType() === 'react-native' && typeof value == 'string') {
+        _value = value;
+      } else if (typeof value !== 'string') {
+        // it's weird but TS forced me to write this much code where I could have just done "getPlatformType() === 'react-native' ? value : value?.target.value" :(
```

Comment:
Remove this? Or just keep it short, something like "Need this to avoid a TS error"

Reply:
that's my love-hate relationship with TS so wanted to express that 😄